File: CS.NET/PdfViewerCSharpAPI_R2/DocumentManagement/Requests/PdfGetOutlinesRequest.cs
// -----------------------------------------------------------------------
// <copyright file="PdfGetOutlinesRequest.cs" company="">
// TODO: Update copyright text.
// </copyright>
// -----------------------------------------------------------------------

namespace PdfTools.PdfViewerCSharpAPI.DocumentManagement.Requests
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using PdfTools.PdfViewerCSharpAPI.Model;
    using PdfTools.PdfViewerCSharpAPI.Utilities;

    /// <summary>
    /// TODO: Update summary.
    /// </summary>
    public class PdfGetOutlinesRequest : APdfRequest<int, IList<PdfOutlineItem>>
    {

        public PdfGetOutlinesRequest(int outlineId)
            :base(outlineId, 72)
        {
        }

        protected override IList<PdfOutlineItem> ExecuteNative(IPdfDocument document, int outlineId)
        {
            return document.GetOutlines(outlineId);       
        }

        protected override void triggerControllerCallback(IPdfControllerCallbackManager controller, InOutTuple tuple, PdfViewerException ex)
        {
            controller.OnOutlinesLoaded(ex, tuple.arguments, tuple.output);
        }
        protected override void triggerControllerCallback(IPdfControllerCallbackManager controller, PdfViewerException ex)
        {
            controller.OnOutlinesLoaded(ex, this.arguments, null);
        }
    }
}

File: Java/PdfViewerAPI_R2/java/com/pdf_tools/pdfviewer/Requests/PdfGetOutlinesRequest.java
package com.pdf_tools.pdfviewer.Requests;

import com.pdf_tools.pdfviewer.DocumentManagement.IPdfDocument;
import com.pdf_tools.pdfviewer.Model.IPdfControllerCallbackManager;
import com.pdf_tools.pdfviewer.Model.PdfOutlineItem;
import com.pdf_tools.pdfviewer.Model.PdfViewerException;

public class PdfGetOutlinesRequest extends APdfRequest<PdfOutlineItem[]>
{

    public PdfGetOutlinesRequest(int parentId)
    {
        this._priority = 9;
        this.parentId = parentId;
    }

    @Override
    public void execute(IPdfDocument document, IPdfControllerCallbackManager controller)
    {
        try
        {
            PdfOutlineItem items[] = document.getOutlines(parentId);
            controller.onOutlinesLoaded(parentId, items, null);
            completedEvent.triggerEvent(items, null);
        } catch (PdfViewerException ex)
        {
            controller.onOutlinesLoaded(parentId, null, ex);
            completedEvent.triggerEvent(null, ex);
        }
    }

    private int parentId;

}
